server/src/test/java/org/elasticsearch/search/internal/ContextIndexSearcherTests.java

```diff
@@ -376,6 +379,54 @@ public void onRemoval(ShardId shardId, Accountable accountable) {
         IOUtils.close(reader, w, dir);
     }
 
+    public void testComputeSlices() throws IOException {
+        Directory dir = newDirectory();
+        IndexWriter w = new IndexWriter(dir, newIndexWriterConfig(null));
+        int numDocs = randomIntBetween(1000, 25000);
+        Document doc = new Document();
+        for (int i = 0; i < numDocs; i++) {
+            w.addDocument(doc);
+            if (frequently()) {
+                // lets create many segments
+                w.commit();
+            }
+        }
+        {
+            DirectoryReader reader = DirectoryReader.open(w);
+            List<LeafReaderContext> contexts = reader.leaves();
+            for (int numThreads = 1; numThreads < 16; numThreads++) {
```

Comment:
maybe use some random number instead of that exhaustive loop and loop over a random number as iterations. This might catch issues when the same number is use multiple times etc.

Reply:
Changed for a random number of iterations